test/integration/controller/admissionchecks/provisioning/provisioning_test.go

```diff
@@ -597,6 +597,28 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 					state := workload.FindAdmissionCheck(updatedWl.Status.AdmissionChecks, ac.Name)
 					g.Expect(state).NotTo(gomega.BeNil())
 					g.Expect(state.State).To(gomega.Equal(kueue.CheckStateReady))
+					g.Expect(state.PodSetUpdates).To(gomega.BeComparableTo([]kueue.PodSetUpdate{
+						{
+							Name: "ps1",
+							Annotations: map[string]string{
+								provisioning.ConsumesAnnotationKey: provReqKey.Name,
+							},
+						},
+						{
+							Name: "ps2",
+							Annotations: map[string]string{
+								provisioning.ConsumesAnnotationKey: provReqKey.Name,
+							},
+						},
+					}))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Check the workload is admitted", func() {
```

Comment:
I think this is the actual fix, to make sure that the workload is admitted, so that we don't go beyond this check: https://github.com/kubernetes-sigs/kueue/blob/395137afeaad94179db68043e5c74ab193f6beb1/pkg/controller/admissionchecks/provisioning/controller.go#L138. 

Previously the test would be flaky, because going beyond the condition still required some extra work to evict the workload. Whereas the check in "Checking the workload remains admitted" would be fast. 

Could you please verify this by using `Consistently` instead of `Eventually` here on the main branch? I expect you would get the failure consistently. If you confirm the case, then please use `Consistently` on the fix and confirm that it is stable now. If we can confirm this, I'm ok with either `Eventually` or `Consistently` to be actually merged. The drawback of `Consistently` is that by default it takes 3s, but maybe we can use 1s.

Reply:
`Eventually` passed the test successfully. I will test `Consistently` next.

Reply:
You can also run the tests locally. I would usually do 
1. mark the test I want to run in isolation with `ginkgo.FIt` instead of `ginkgo.It` (F for focus)
2. run integration tests selectively for the package: `INTEGRATION_TARGET=./test/integration/controller/admissionchecks/provisioning make test-integration`

Reply:
Ok. let's use `Eventually`.